include/wil/registry.h

```diff
@@ -162,6 +162,108 @@ namespace wil
         }
 #endif // #define __WIL_WINREG_STL
 
+#if defined(WIL_ENABLE_EXCEPTIONS)

```

Comment:
If I'm reading this correctly, these iterator types are not available to consumers without exceptions, which seems like an unnecessary limitation. We should follow the design of the `IVector*`, etc. iterator types and make them available in throwing, `_nothrow`, and `_failfast` forms.